ambari-common/src/main/python/ambari_commons/repo_manager/yum_manager.py

```diff
@@ -352,7 +330,7 @@ def rpm_check_package_available(self, name):
     regex = re.compile(name_regex)
 
     for package in packages:
-      if regex.match(package['name']):
+      if regex.match(package['name'].decode()):
```

Comment:
This is not working, it is failing with following error
`2023-10-27 19:00:36,138 - Reporting component version failed
Traceback (most recent call last):
  File "/usr/lib/ambari-agent/lib/resource_management/libraries/script/script.py", line 354, in execute
    method(env)
  File "/var/lib/ambari-agent/cache/stack-hooks/before-INSTALL/scripts/hook.py", line 33, in hook
    install_packages()
  File "/var/lib/ambari-agent/cache/stack-hooks/before-INSTALL/scripts/shared_initialization.py", line 37, in install_packages
    retry_count=params.agent_stack_retry_count)
  File "/usr/lib/ambari-agent/lib/resource_management/core/base.py", line 123, in __new__
    cls(names_list.pop(0), env, provider, **kwargs)
  File "/usr/lib/ambari-agent/lib/resource_management/core/base.py", line 164, in __init__
    self.env.run()
  File "/usr/lib/ambari-agent/lib/resource_management/core/environment.py", line 163, in run
    self.run_action(resource, action)
  File "/usr/lib/ambari-agent/lib/resource_management/core/environment.py", line 127, in run_action
    provider_action()
  File "/usr/lib/ambari-agent/lib/resource_management/core/providers/packaging.py", line 30, in action_install
    self._pkg_manager.install_package(package_name, self.__create_context())
  File "/usr/lib/ambari-agent/lib/ambari_commons/repo_manager/yum_manager.py", line 211, in install_package
    elif not self._check_existence(name) or context.action_force:
  File "/usr/lib/ambari-agent/lib/ambari_commons/repo_manager/yum_manager.py", line 279, in _check_existence
    return self.rpm_check_package_available(name)
  File "/usr/lib/ambari-agent/lib/ambari_commons/repo_manager/yum_manager.py", line 333, in rpm_check_package_available
    if regex.match(package['name'].decode()):
AttributeError: 'str' object has no attribute 'decode'`

package['name'] is already unicode_string, no need to decode it

Reply:
Sometimes the return of `package['name']` is not `str`, so I think we need to add a judgement this place.

Reply:
> Sometimes the return of `package['name']` is not `str`, so I think we need to add a judgement this place.

yes, please add it

Reply:
@wenwj0  seems you have not addressed this comment.
you can change the code with some thing like this:
` pkg_name = package['name'].decode() if isinstance(package['name'], bytes) else package['name']
      if regex.match(pkg_name):`